Multilateration improvments

jaxnb opened the pull request:
closes #148

jaxnb committed:
Improved multilateration script

jaxnb committed:
Improving multilateration script. Ready for testing; lots of mess at the moment

jaxnb committed:
Multilateration filtering working. Documented

kev-the-dev commented on drivers/mil_passive_sonar/scripts/multilateration:
```diff
@@ -24,30 +24,40 @@ class MultilaterationNode(object):
         self.enabled = False
-        self.global_fame = 'enu'
+        self.last_pos = None
+        self.global_frame = 'enu'
```
As this will be a mil_common script (likely to be used on Subjugator), the global frame should also be a param (Subjugator uses "map")

jaxnb replied:
Done

kev-the-dev commented on drivers/mil_passive_sonar/scripts/multilateration:
```diff
@@ -57,25 +67,82 @@ class MultilaterationNode(object):
+                if distance1 < self.filter_dist or distance2 < self.filter_dist:
+                    line_intersections += 1
+
+            if line_intersections < self.filter_dist:
```
Should this be `self.filter_count`?

jaxnb replied:
Good catch! I've fixed this now. I guess it passed my testing since the default values are close.

kev-the-dev commented on drivers/mil_passive_sonar/scripts/multilateration:
```diff
@@ -57,25 +67,82 @@ class MultilaterationNode(object):
+        line_array = np.array([[0, 0, 0, 0]])
+        line_intersecting_array = np.array([])
+
+        for line1 in lines_array:
```
Comment noting that this section filters out nearby intersections

jaxnb replied:
Done

kev-the-dev commented on drivers/mil_passive_sonar/scripts/multilateration:
```diff
@@ -57,25 +67,82 @@ class MultilaterationNode(object):
+
+        # If no point could be determined, don't publish
+        if where is None:
+            return
```
This is a nice place to do a `rospy.logwarn_throttle(10.0, "could not ...")`. So if nothing gets published the logs show why

jaxnb replied:
Done

kev-the-dev reviewed (changes requested):
Well done overall. Have a couple q's

jaxnb committed:
Fix Kevin's comments
 - Generalized global frame
 - Bug with filter
 - Additional documentation
 - Additional feedback
